src/rdiff_backup/connection.py

```diff
@@ -363,6 +366,23 @@ def _close(self):
         """Close the pipes associated with the connection"""
         self.outpipe.close()
         self.inpipe.close()
+        # reap the pipe child with wait() to ensure any final output from
+        # the pipe by commands that run after rdiff-backup server; otherwise
+        # a race condition occurs where final output is sometimes lost;
+        # Python>=3.3 gives the timeout option to wait: set to a modestly
+        # small value here to minimize possibility of introducing bugs/delays
+        if self.process:
+            try:
+                self.process.wait(timeout=5)
```

Comment:
I think an INFO level message of the kind `"Server process {sp} has exited with return code {rc}".format(sp=self.process.args, rc=self.process.returncode)` could be helpful in certain circumstances.